### PR TITLE
Replace repetitive actions with Action Groups in AdminDeleteSimpleProductTest

### LINKED ISSUE #33783: [Issue] Replace repetitive actions with Action Groups in AdminDeleteSimpleProductTest
This issue is automatically created based on existing pull request: magento/magento2#31963: Replace repetitive actions with Action Groups in AdminDeleteSimpleProductTest

---------

### Description (*)
The test is refactored according to the best practices followed by MFTF.

### Manual testing scenarios (*)
Run the test in the local environment and verified that it run successfully.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### PR DESCRIPTION
**Pull request by kate-kyzyma**
### Description (*)
The test is refactored according to the best practices followed by MFTF.

### Manual testing scenarios (*)
Run the test in the local environment and verified that it run successfully.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### Resolved issues:
1. [x] resolves magento/magento2#33783: Replace repetitive actions with Action Groups in AdminDeleteSimpleProductTest

### CODE REVIEW

**Comment by kate-kyzyma** [comment]
@magento run all tests

File: app/code/Magento/CatalogSearch/Test/Mftf/ActionGroup/AssertStorefrontProductNotOnSearchPageActionGroup.xml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+<!--
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+
+<actionGroups xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+              xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/actionGroupSchema.xsd">
+
```
Comment (dmytro-ch) [inline app/code/Magento/CatalogSearch/Test/Mftf/ActionGroup/AssertStorefrontProductNotOnSearchPageActionGroup.xml]:
```suggestion
```

File: app/code/Magento/CatalogSearch/Test/Mftf/ActionGroup/AssertStorefrontProductNotOnSearchPageActionGroup.xml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    <actionGroup name="AssertStorefrontProductNotOnSearchPageActionGroup">
+        <arguments>
+            <argument name="productSku" type="string"/>
+        </arguments>
+
```
Comment (dmytro-ch) [inline app/code/Magento/CatalogSearch/Test/Mftf/ActionGroup/AssertStorefrontProductNotOnSearchPageActionGroup.xml]:
```suggestion
```

File: app/code/Magento/CatalogSearch/Test/Mftf/ActionGroup/AssertStorefrontNoResultsMessageOnSearchPageActionGroup.xml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+
+<actionGroups xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+              xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/actionGroupSchema.xsd">
+
+    <actionGroup name="AssertStorefrontNoResultsMessageOnSearchPageActionGroup">
+
```
Comment (dmytro-ch) [inline app/code/Magento/CatalogSearch/Test/Mftf/ActionGroup/AssertStorefrontNoResultsMessageOnSearchPageActionGroup.xml]:
```suggestion
```

File: app/code/Magento/CatalogSearch/Test/Mftf/ActionGroup/AssertStorefrontNoResultsMessageOnSearchPageActionGroup.xml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+<!--
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+
+<actionGroups xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+              xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/actionGroupSchema.xsd">
+
```
Comment (dmytro-ch) [inline app/code/Magento/CatalogSearch/Test/Mftf/ActionGroup/AssertStorefrontNoResultsMessageOnSearchPageActionGroup.xml]:
```suggestion
```

File: app/code/Magento/CatalogSearch/Test/Mftf/ActionGroup/AssertStorefrontProductNotOnSearchPageActionGroup.xml [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+
+<actionGroups xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+              xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/actionGroupSchema.xsd">
+
+    <actionGroup name="AssertStorefrontProductNotOnSearchPageActionGroup">
```
Comment (dmytro-ch) [inline app/code/Magento/CatalogSearch/Test/Mftf/ActionGroup/AssertStorefrontProductNotOnSearchPageActionGroup.xml]:
```suggestion
    <actionGroup name="AssertStorefrontProductIsMissingOnSearchResultPageActionGroup">
```

**Review by dmytro-ch — CHANGES_REQUESTED** [review]
Hello @kate-kyzyma, could you please check and fix the issue reported by `Semantic Version Checker`?

File: app/code/Magento/CatalogSearch/Test/Mftf/ActionGroup/AssertStorefrontProductNotOnSearchPageActionGroup.xml (hunk 1 above)
Comment (kate-kyzyma) [reply app/code/Magento/CatalogSearch/Test/Mftf/ActionGroup/AssertStorefrontProductNotOnSearchPageActionGroup.xml]: Fixed

File: app/code/Magento/CatalogSearch/Test/Mftf/ActionGroup/AssertStorefrontProductNotOnSearchPageActionGroup.xml (hunk 2 above)
Comment (kate-kyzyma) [reply app/code/Magento/CatalogSearch/Test/Mftf/ActionGroup/AssertStorefrontProductNotOnSearchPageActionGroup.xml]: Fixed

File: app/code/Magento/CatalogSearch/Test/Mftf/ActionGroup/AssertStorefrontNoResultsMessageOnSearchPageActionGroup.xml (hunk 3 above)
Comment (kate-kyzyma) [reply app/code/Magento/CatalogSearch/Test/Mftf/ActionGroup/AssertStorefrontNoResultsMessageOnSearchPageActionGroup.xml]: Fixed

File: app/code/Magento/CatalogSearch/Test/Mftf/ActionGroup/AssertStorefrontNoResultsMessageOnSearchPageActionGroup.xml (hunk 4 above)
Comment (kate-kyzyma) [reply app/code/Magento/CatalogSearch/Test/Mftf/ActionGroup/AssertStorefrontNoResultsMessageOnSearchPageActionGroup.xml]: Fixed

File: app/code/Magento/CatalogSearch/Test/Mftf/ActionGroup/AssertStorefrontProductNotOnSearchPageActionGroup.xml (hunk 5 above)
Comment (kate-kyzyma) [reply app/code/Magento/CatalogSearch/Test/Mftf/ActionGroup/AssertStorefrontProductNotOnSearchPageActionGroup.xml]: Fixed

**Comment by kate-kyzyma** [comment]
Hello @dmytro-ch!
I have fixed the mentioned issues. Please, check.

**Comment by kate-kyzyma** [comment]
@magento run Semantic Version Checker

**Comment by kate-kyzyma** [comment]
@magento run Semantic Version Checker

**Comment by kate-kyzyma** [comment]
@magento run all tests

**Comment by dmytro-ch** [comment]
@magento run Static Tests, WebAPI Tests, Functional Tests B2B, Functional Tests EE

**Comment by dmytro-ch** [comment]
@magento run Static Tests

**Comment by dmytro-ch** [comment]
@magento run Static Tests, Functional Tests B2B,

**Comment by dmytro-ch** [comment]
@magento run Functional Tests B2B, Functional Tests EE

**Comment by kate-kyzyma** [comment]
@magento run Functional Tests B2B, Functional Tests EE

**Comment by kate-kyzyma** [comment]
@magento run all tests

**Comment by kate-kyzyma** [comment]
@magento run Functional Tests B2B, Functional Tests EE

**Comment by kate-kyzyma** [comment]
@magento run Functional Tests B2B

**Comment by dmytro-ch** [comment]
@magento run Functional Tests B2B

**Comment by kate-kyzyma** [comment]
@magento run all tests

**Comment by kate-kyzyma** [comment]
@magento run Functional Tests EE, WebAPI Tests

**Comment by kate-kyzyma** [comment]
@magento run Functional Tests EE

**Comment by kate-kyzyma** [comment]
@magento run all tests

**Comment by dmytro-ch** [comment]
@magento run Integration Tests

**Review by dmytro-ch — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @dmytro-ch, thank you for the review.
[ENGCOM-9048](https://jira.corp.magento.com/browse/ENGCOM-9048) has been created to process this Pull Request

**Comment by eduard13** [comment]
@magento create issue

**Review by eduard13 — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @eduard13, thank you for the review.
[ENGCOM-9048](https://jira.corp.magento.com/browse/ENGCOM-9048) has been created to process this Pull Request

**Comment by engcom-Charlie** [comment]
✔️ QA Passed

The provided MFTF test case is working fine and green in build. Please find the Allure report for the same.

image

image

**Pull request merged**